docs/src/resources/two-side-column.md

```diff
@@ -0,0 +1,519 @@
+}
+```
+</td>
+             <td>Text comes here</td>
+</tr>
+<tr>
+<td>
+
+```
+module TestBench;
+wire [31:0] sum;
+ParameterizedWidthAdder  #(32, 32, 32) my32BitAdderWithTruncation (32'b0, 32'b0, sum);
+endmodule
+```
+</td>
+            <td>
+
+```
+val my32BitAdderWithTruncation = Module(new ParameterizedWidthAdder(32, 32, 32)
+```
+</td>
+             <td>Text comes here</td>
+         </tr>
+ <tr>
+<td>
+Text comes here
+</td>
+<td>
+
+```scala mdoc:silent
+class MyModule extends Module {
+  val io = IO(new Bundle {
+    val in  = Input(UInt(4.W))
+    val out = Output(UInt(4.W))
+  })
+
+val two  = 1 + 1
+println(two)
+val utwo = 1.U + 1.U
+println(utwo)
+
+io.out := io.in
+}
+println(getVerilogString(new MyModule))
+```
+</td>
+<td>
+
+```
+2
+UInt<1>(OpResult in MyModule)
+module MyModule(
+  input        clock,
+  input        reset,
+  input  [3:0] io_in,
+  output [3:0] io_out
+);
+  assign io_out = io_in; // @[main.scala 19:10]
+endmodule
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+#Wire assignment
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+            <td><b style="font-size:30px">Generated Verilog</b></td>
+         </tr>
+         <tr>
+<td>Text comes here</td>
+<td>
+
+```scala mdoc:silent
+class MyModule extends Module {
+val io = IO(new Bundle {
+val in = Flipped(DecoupledIO(UInt(8.W)))
+val out = DecoupledIO(UInt(8.W))
+})
+
+val tmp = Wire(DecoupledIO(UInt(8.W)))
+tmp <> io.in
+io.out <> tmp
+io.out <> io.in
+}
+
+
+println(getVerilogString(new MyModule))
+```
+</td>
+<td>
+
+```
+module MyModule(
+  input        clock,
+  input        reset,
+  output       io_in_ready,
+  input        io_in_valid,
+  input  [7:0] io_in_bits,
+  input        io_out_ready,
+  output       io_out_valid,
+  output [7:0] io_out_bits
+);
+  assign io_in_ready = io_out_ready; // @[main.scala 17:12]
+  assign io_out_valid = io_in_valid; // @[main.scala 17:12]
+  assign io_out_bits = io_in_bits; // @[main.scala 17:12]
+endmodule
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+#Register assignment
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+            <td><b style="font-size:30px">Generated Verilog</b></td>
+         </tr>
+         <tr>
+<td>Text comes here</td>
+<td>
+
+```scala mdoc:silent
+class RegisterModule extends Module {
+  val io = IO(new Bundle {
+    val in  = Input(UInt(12.W))
+    val out = Output(UInt(12.W))
+  })
+
+val register = Reg(UInt(12.W))
+register := io.in + 1.U
+io.out := register
+}
+
+println(getVerilogString(new RegisterModule))
+```
+</td>
+<td>
+
+```
+module RegisterModule(
+  input         clock,
+  input         reset,
+  input  [11:0] io_in,
+  output [11:0] io_out
+);
+`ifdef RANDOMIZE_REG_INIT
```

Comment:
@sequencer didn't you recently commit somethign that would allow us to not emit all this `RANDOMIZE` stuff? Can you point me to the PR or suggest how to call `emitVerilog` to not have all this stuff here, to make the verilog more canonical?

Reply:
See:
https://github.com/chipsalliance/firrtl/pull/2416
https://github.com/chipsalliance/firrtl/pull/2449

Reply:
I think `emitVerilog` needs additional API to add annotation. Maybe we need to add additional feature to this.

Reply:
`emitVerilog` already supports arguments, it can be used as:

```
println((new ChiselStage).emitVerilog((new Example), Array("--emission-options=disableRegisterRandomization,disableMemRandomization")))
```

I've submitted a PR adding the same functionality to `getVerilogString` in #2401

Reply:
I think it's irrelevant for this PR since we aren't going to show the emitted verilog, so I will resolve this comment